src/desert/_make.py

```diff
@@ -370,6 +377,18 @@ def _get_field_default(
         raise TypeError(field)
 
 
+def _is_typeddict(typ: t.Any) -> bool:
+    # typing_inspect misses some case.
+    # python>=3.10: use t.is_typeddict
+    if hasattr(t, "is_typeddict"):
+        return t.is_typeddict(typ)
+    # python>=3.8; <3.10: Reimplement t.is_typeddict
+    if hasattr(t, "_TypedDictMeta"):
+        return isinstance(typ, t._TypedDictMeta)  # type: ignore[attr-defined]
+    # Fallback to typing_inspect
+    return typing_inspect.typed_dict_keys(typ) is not None
```

Comment:
Let's use versions here to pick.  More hint checkable and less prone to using the wrong thing due to non-obvious typos.

```python3
if sys.version_info >= (3, 10):
elif sys.version_info >= (3, 8):
else:
```